apps/application-system/api/src/app/modules/application/lifecycle/application-lifecycle.service.ts

```diff
@@ -0,0 +1,165 @@
+import { Inject, Injectable } from '@nestjs/common'
+import { APPLICATION_CONFIG } from '../application.configuration'
+import type { ApplicationConfig } from '../application.configuration'
+import { Application } from '../application.model'
+import { ApplicationService } from '../application.service'
+import { AwsService } from '../files'
+import AmazonS3URI from 'amazon-s3-uri'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import type { Logger } from '@island.is/logging'
+
+export interface ApplicationPruning {
+  pruned: boolean
+  application: Application
+  failedAttachments: object
+}
+
+@Injectable()
+export class ApplicationLifeCycleService {
+  private processingApplications: ApplicationPruning[] = []
+
+  constructor(
+    @Inject(APPLICATION_CONFIG)
+    private readonly config: ApplicationConfig,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
```

Comment:
We might want to create a child logger with its own context to make these easier to query. Something like:

```
    this.logger = logger.child({ context: 'ApplicationLifeCycleService' })
```